### PR TITLE
Adding Checkup spec

### PR DESCRIPTION
**Pull request by scalvert**
[Rendered](https://github.com/checkupjs/checkup/blob/spec/SPEC.md)

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+It's likely that we'll want to use specific lint rules both in Checkup and configured to run in a project. Again, Checkup's role is one of a data aggregator, and lint rules provide more fine-grained, actionable guidance on changes within source code.
+
+### Results
+
+`Tasks` return `Results`, which are in turn formatted for output based on what data they return.
+
+### Output
+
+`Results` are aggregated from `Tasks`, formatted, and output depending on their target. Output can be any of `console`, `PDF`, or `HTML`.
```
Comment (lbdm44) [inline SPEC.md]: Should we add `JSON` to the output list?

File: SPEC.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+- Organization specific
+
+  Any plugins that are specific to an organization may not make sense to include as part of checkup directly. Ensuring that specific internal APIs are used the correct way, for instance, may be a concern best left to consuming organizations.
+
+- External library specific
+
+  Plugins that are specific to external libraries, such as `ember-concurrency` for example, may be best left as externally authored plugins that are ultimately configurable.
+
+## Configuration
```
Comment (mahirshah) [inline SPEC.md]: I imagine each project would maintain a .checkuprc file or similar in order to configure which Tasks to run and the config options passed to those Tasks?

File: SPEC.md (hunk 1 above)
Comment (scalvert) [reply SPEC.md]: Good call. Ya I think we should add it.

File: SPEC.md (hunk 2 above)
Comment (scalvert) [reply SPEC.md]: Agree, which is what our original thought was. Either that or package.json (or both!).

I wanted to keep those kinds of details out of this spec though, as this is more the high level requirements.

File: SPEC.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+### Tasks
+
+`Tasks` are the main building block of Checkup. They should provide a thin, simple interface that allows consumers to author tasks in the most straightforward manner.
+
+Tasks fall into one of two categories:
+
+#### `JSON`
+
+A JSON task is used to read JSON files, whether project information files (such as `project.json`), or configuration files (such as `.eslinrc.json` and the like). These tasks simply gather information from disparate sources to aggregate into a report.
```
Comment (suchitadoshi1987) [inline SPEC.md]: this is a totally valid example, but should we use the `package.json` as an example instead?

Comment (scalvert) [reply SPEC.md]: Derp. Thanks for catching this!

**Comment by scalvert** [comment]
@rwjblue @lbdm44 @suchitadoshi1987 @mahirshah 

If none of you have any comments as to the overall high-level strategy outlined in this spec, I'm going to merge it by EOD Friday. We can always amend/update it after - it's not set in stone.

**Comment by lbdm44** [comment]
LGTM

**Pull request merged**